### PR TITLE
add daemon to monitor jenkins job and trigger the hasal framework

### PR DESCRIPTION
**Pull request by ShakoHo**
@ypwalter @askeing @MikeLien r?

### CODE REVIEW

File: agent/agent.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+        logger.info("Initialing Main Runner for Hasal agent")
+        if os.path.isdir(dirpath):
+            self.dirpath = dirpath
+            logger.info("loading runner config folder: " + dirpath)
+        else:
+            logger.info(dirpath + " is invalid, use default path instead")
+
+        self.scheduler = BackgroundScheduler()
+        self.scheduler.start()
+        self.load_dir(dirpath)
```
Comment (MikeLien) [inline agent/agent.py]: should be `self.dirpath`

**Comment by MikeLien** [comment]
lgtm r+

File: agent/agent.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+        event_handler = JsonHandler(patterns=["*.json"], ignore_directories=True)
+        event_handler.set_handler(oncreated=self.load, onmodified=self.load, ondeleted=self.remove)
+        observer = Observer()
+        observer.schedule(event_handler, self.dirpath, recursive=True)
+        observer.start()
+
+    def load_dir(self, folder):
+        (dirpath, dirnames, filenames) = os.walk(folder).next()
```
Comment (ypwalter) [inline agent/agent.py]: folder should be checked first to prevent from exception

**Comment by ypwalter** [comment]
tested setup requirements and went through code changed.
r+

**Pull request merged**